internal/cli/deployment_list.go

```diff
@@ -95,34 +95,26 @@ func (c *DeploymentListCommand) Run(args []string) int {
 		}
 
 		// List builds
-		resp, err := client.ListDeployments(c.Ctx, &pb.ListDeploymentsRequest{
+		resp, err := client.UI_ListDeployments(c.Ctx, &pb.UI_ListDeploymentsRequest{
 			Application:   app.Ref(),
 			Workspace:     wsRef,
 			PhysicalState: phyState,
 			Status:        c.filterFlags.statusFilters(),
-			Order:         c.filterFlags.orderOp(),
-			LoadDetails:   pb.Deployment_BUILD,
 		})
 		if err != nil {
+			if s, ok := status.FromError(err); ok {
+				if s.Code() == codes.Unimplemented {
+					c.project.UI.Output(
+						"This CLI version is incompatible with the current server - missing UI_ListDeployments method. Upgrade your server or downgrade your CLI.",
```

Comment:
I would mention the version of the server that you need. You can also use `c.project.ServerVersion()` I think to get the current server version (its cached, the client gets it on initialiization) if you want to show that to the user.